datasets/colours.py

```diff
@@ -31,10 +31,21 @@
 ]
 
 
+def match_color(text, colors):
+    """
+    Check if any of the exact words in 'colors' is in the text.
+    Returns 1 if any color is found as a whole word, otherwise 0.
+    """
+    colors_pattern = '|'.join([re.escape(color) for color in colors])
+    pattern = rf'(?<!\S)(?:{colors_pattern})(?!\S)|(?<!\S)(?:{colors_pattern})(?=[,.!?])'
+    match = re.search(pattern, text, re.IGNORECASE)
+    return 1 if match else 0
```

Comment:
Why is it not returning a `bool`?

Reply:
Originally, the result was used as a conditional but is no longer so can probably be cleaned